code/modules/mob/living/silicon/robot/robot.dm

```diff
@@ -653,10 +653,15 @@ var/list/robot_verbs_default = list(
 					var/datum/robot_component/C = components[V]
 					if(C.installed == 1 || C.installed == -1)
 						removable_components += V
-
+				if(src.module && istype(src.module, /obj/item/robot_module/miner))
+					removable_components += "KA modkits"
 				var/remove = input(user, "Which component do you want to pry out?", "Remove Component") as null|anything in removable_components
 				if(!remove)
 					return
+				if(remove == "KA modkits")

```

Comment:
I'm not a big fan of implementing custom "removal" aspects like this - instead, I would prefer a system as follows:
1 new var on robot modules: `var/list/custom_removals = list()`
1 new proc on robot modules: 
```
// Return true in an overridden subtype to prevent normal removal handling
/obj/item/robot_module/proc/handle_custom_removal(component_id, mob/living/user, obj/item/W, params)
    return FALSE
```

And then in `robot_module/miner`, override that proc to be as follows:
```
/obj/item/robot_module/miner
    // var declarations...
    custom_removals = list("KA modkits")
// other code...
/obj/item/robot_module/miner/handle_custom_removal(component_id, mob/living/user, obj/item/W, params)
    if(component_id == "KA modkits")
        for(var/obj/item/gun/energy/kinetic_accelerator/cyborg/D in src)
            D.attackby(W, user, params)
        return TRUE
    return ..()
```

Finally, hook this up to the rest of the code by replacing the following:
Lines 656-657 with:
`removable_components += module.custom_removals`
Lines 661-664 with:
```
if(module.handle_custom_removal(remove, user, W, params))
    return
```